core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -793,10 +839,25 @@ class HttpGoogleServicesDAO(
       })
 
       // enable appropriate google apis
-      _ <- Future.sequence(projectTemplate.services.map { service => retryExponentially(when500orGoogleError)(() => {
-        enableServiceApi(projectName.value, service, credential)
-      })})
+      operations <- Future.sequence(projectTemplate.services.map { service => retryWhen500orGoogleError(() => {
+        executeGoogleRequest(serviceManager.services().enable(service, new EnableServiceRequest().setConsumerId(s"project:${projectName.value}")))
+      }) map { googleOperation =>
+        RawlsBillingProjectOperationRecord(projectName.value, service, googleOperation.getName, toScalaBool(googleOperation.getDone), Option(googleOperation.getError).map(error => toErrorMessage(error.getMessage, error.getCode)), API_SERVICE_MANAGEMENT)
+      }})
 
+    } yield {
+      operations
+    })
+  }
+
+  override def completeProjectSetup(project: RawlsBillingProject): Future[Try[Unit]] = {
```

Comment:
It's unclear to me what the division is between `beginProjectSetup` and `completeProjectSetup` such that, if I were to add something, I wouldn't be sure where it should happen. Could you add some docs to explain the difference?